pyproject.toml

```
@@ -1,19 +1,117 @@
-[tool.black]
-exclude = "3rdparty/*"
-line-length = 100
+[project]
+name = "xgrammar"
+description = "Efficient, Flexible and Portable Structured Generation"
+authors = [{ name = "MLC Team" }]
+readme = "README.md"
+license = { text = "Apache 2.0" }
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Education",
+    "Intended Audience :: Science/Research",
+]
+keywords = ["machine learning", "inference"]
+requires-python = ">=3.9, <4"
+dependencies = [
+    "pydantic",
+    "sentencepiece",
+    "tiktoken",
+    "torch>=2.5.0",
```

Support old versions of pytorch
```suggestion
    "torch>=1.10.0",
```